web_src/js/components/RepoContributors.vue

```diff
@@ -27,6 +28,8 @@ import 'chartjs-adapter-dayjs-4/dist/chartjs-adapter-dayjs-4.esm';
 import {fomanticQuery} from '../modules/fomantic/base.ts';
 import type {Entries} from 'type-fest';
 
+const {pageData} = window.config;
```

Comment:
It doesn't need to use "pageData" here.

You could pass the `repoDefaultBranch` in the same way as "repoLink"

Reply:
I could not find a way to reference the `ctx.Repo.RefName` as the `repoLink` did in the file `web_src/js/features/contributors.ts` and the `'#repo-contributors-chart'` div does not have the data property to retrieve the default branch name.

Do you have a guide to improve this usage? I found some other components that use `pageData` as well.

Reply:
The `pageData` was designed and introduced by me to resolve some legacy refactoring problems (#17301), it is only an intermediate solution, if it could be avoided, I'd like to prefer to avoid it.

I will make some changes in this PR.

Reply:
Overall it could be like this 27ee1a2252b4c7c3606d4c84abb1b6b6d3bf88e0